usaspending_api/search/v2/elasticsearch_helper.py

```diff
@@ -66,8 +71,76 @@ def search_transactions(filters, fields, sort, order, lower_limit, limit):
     index_name = index_name[:-1]+'*'
     try:
         response = CLIENT.search(index=index_name, body=query)
-    except Exception as es1:
+    except Exception:
         return False, -1
     total = response['hits']['total']
     results = format_for_frontend(response['hits']['hits'])
-    return results, total
+    return results, total, transaction_type
+
+
+def get_total_results(keyword, index_name):
+    index_name = '{}-{}'.format(TRANSACTIONS_INDEX_ROOT,
+                                index_name.replace('_', ''))+'*'
+    query = {
+        'query': {
+            'query_string': {
+                'query': keyword
+                }
+            }
+    }
+    try:
+        response = CLIENT.search(index=index_name, body=query)
+        return response['hits']['total']
+    except Exception:
+        return -1
+
+
+def spending_by_transaction_count(filters):
+    keyword = filters['keyword']
+    response = {}
+    for category in award_categories:
+        response[category] = get_total_results(keyword, category)
+    return response
+
+
+def search_keyword_id_list_all(keyword):
```

Comment:
Would it be easier for both this function and the download function if this was a generator function instead of iteratively appending the transaction ids to a (potentially very large) python list?

Reply:
Yep, Daniel wants an obnoxiously large python list returned. It's not a great method but as long as the heap memory on the server can handle a request like that whenever someone wants to potentially download something that's fine. Can you let me know what the runtime is for this function.

Reply:
@dpb-bah Can you test this function and comment about runtime?